plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -57,11 +57,12 @@ def keycloak_argument_spec():
     return dict(
         auth_keycloak_url=dict(type='str', aliases=['url'], required=True, no_log=False),
         auth_client_id=dict(type='str', default='admin-cli'),
-        auth_realm=dict(type='str', required=True),
+        auth_realm=dict(type='str'),
         auth_client_secret=dict(type='str', default=None, no_log=True),
-        auth_username=dict(type='str', aliases=['username'], required=True),
-        auth_password=dict(type='str', aliases=['password'], required=True, no_log=True),
-        validate_certs=dict(type='bool', default=True)
+        auth_username=dict(type='str', aliases=['username']),
+        auth_password=dict(type='str', aliases=['password'], default=None, no_log=True),
```

Comment:
Isn't a string variable `None` by default? What's the use of specifying it explicitly?
```suggestion
        auth_password=dict(type='str', aliases=['password'], no_log=True),
```

Reply:
This has been addressed before as well:
https://github.com/ansible-collections/community.general/pull/2250#discussion_r615242861